lib/spack/spack/filesystem_view.py

```diff
@@ -188,47 +188,43 @@ def __init__(self, root, layout, **kwargs):
 
```

Comment:
The changes here seem qualitatively different than in the PR description although perhaps thematically related.: could you add an explanation to the PR description?

Reply:
I made a change to whether empty `projections.yaml` files are written to disk. In the process I cleaned up some spaghetti code to make it easier to make the changes I needed. In the end I think the refactor was more substantial than the changes and it's easy to lose track of the changes in the refactor.